### PR TITLE
Add support for alternate Velocity template directory.

### PR DESCRIPTION
**Pull request by ryonday**
This can be used to "fake" inheritance and interface implementation with the generated Avro classes.

### CODE REVIEW

**Comment by davidmc24** [comment]
Thanks for the pull request. I'll review this tomorrow.

File: src/main/java/com/commercehub/gradle/plugin/avro/GenerateAvroJavaTask.java [hunk 1]
```diff
@@ -39,6 +46,7 @@ public void setEncoding(String encoding) {
 
     @Input
     public String getStringType() {
+        System.out.println("StringType was get: " + stringType);
```
Comment (ryonday) [inline src/main/java/com/commercehub/gradle/plugin/avro/GenerateAvroJavaTask.java]: oops :(

Comment (davidmc24) [inline src/main/java/com/commercehub/gradle/plugin/avro/GenerateAvroJavaTask.java]: No need to fix that.  I'll handle it in a round of minor touch-ups I was going to do anyway.

Comment (ryonday) [inline src/main/java/com/commercehub/gradle/plugin/avro/GenerateAvroJavaTask.java]: Thanks! I added it while trying to debug; I was trying to figure out why the getStringType etc methods seem to have some "magic" to them that I still don't understand. There's one more compiler setting (createSetters) that I'd be glad to implement as well to have a complete feature set. 

Great plugin; we use it extensively here at Under Armour!

**Comment by davidmc24** [comment]
Looks good.  Thanks again for the contribution.  I'm going to add some test coverage for the `templateDirectory` setting.  After I've got that in, it would be great if you could take a pass at `createSetters` support.

**Pull request merged**